File: Lab04/FragmentLambda_Java/app/src/main/java/com/example/fragmentlambda_java/UserDetailsFragment.java
package com.example.fragmentlambda_java;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class UserDetailsFragment extends Fragment {

    private static final String ARG_USER = "user";
    private User user;

    public static UserDetailsFragment newInstance(User user) {
        UserDetailsFragment fragment = new UserDetailsFragment();
        Bundle args = new Bundle();
        args.putParcelable(ARG_USER, user);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            user = getArguments().getParcelable(ARG_USER);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_user_details, container, false);
        TextView txtUserDetails = view.findViewById(R.id.txtUserDetails);

        if (user != null) {
            txtUserDetails.setText("Username: " + user.getUsername() + "\nPassword: " + user.getPassword());
        }

        return view;
    }
}

File: FragmentLambda/app/src/main/java/com/example/fragmentlambda/UserDetailsFragment.kt
package com.example.fragmentlambda

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

private const val ARG_USER = "user"

class UserDetailsFragment : Fragment() {

    private var user: User? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            user = it.getParcelable(ARG_USER)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_user_details, container, false)
        val txtUserDetails = view.findViewById<TextView>(R.id.txtUserDetails)

        user?.let {
            txtUserDetails.text = "Username: ${it.username}\nPassword: ${it.password}"
        }

        return view
    }

    companion object {
        @JvmStatic
        fun newInstance(user: User?) =
            UserDetailsFragment().apply {
                arguments = Bundle().apply {
                    putParcelable(ARG_USER, user)
                }
            }
    }
}
